tasks/connectors/aqua/lib/aqua_helper.rb

```diff
@@ -21,14 +37,14 @@ def get_token_from_on_prem(aqua_url, username, password)
       end
 
       def get_token_from_cloud(username, password)
-        get_token(SAAS_AUTH_URL, username, password)
+        get_token(@saas_auth_url, username, password)
       end
 
       def get_token(auth_url, username, password)
         print_debug "Getting Auth Token from #{auth_url}"
 
         headers = { "Content-Type" => "application/json" }
-        payload = if auth_url == SAAS_AUTH_URL
+        payload = if auth_url == @saas_auth_url
```

Comment:
Instead of comparing strings to figure out if this is cloud-based or on-prem, I'd suggest doing a check on initialization and setting an instance variable that has a reader method. Maybe `cloud?` or `saas?`. That way a typo in one of the strings in your case statement won't prevent this check from working. It could be a headache to debug why saas won't auth with an id param but only in one region.